tests/suites/os/tests/secureboot/index.js

```diff
@@ -1,145 +1,339 @@
+			this.worker.executeCommandInHostOS(
+				'balena logs $(balena ps -aqf NAME=load)',
+				this.link,
+			),
+			/Key was rejected by service/,
+			'Unsigned module does not load',
+		);
+	}
+
+	async testBootloaderIntegrity() {
+		throw new Error("Method testBootloaderIntegrity() must be implemented");
+	}
+
+	async testBootloaderConfigIntegrity() {
+		throw new Error("Method testBootloaderConfigIntegrity() must be implemented");
+	}
+}
+
+class uefiSecureBoot extends secureBoot {
+	async isSecureBootSupported() {
+			return this.worker.executeCommandInHostOS(
+				['ls', securebootEfiVarPath, '&>/dev/null',
+					'&&', 'echo', 'true',
+					'||', 'echo', 'false'
+				],
+				this.link,
+			).then(output => { return output === 'true'});
+	}
+
+	async isSecureBootEnabled() {
+			// https://wiki.archlinux.org/title/Unified_Extensible_Firmware_Interface/Secure_Boot#Checking_Secure_Boot_status
+			return this.worker.executeCommandInHostOS(
+					[
+						'od', '--address-radix=n', '--format=u1', securebootEfiVarPath,
+							'|', 'tr', '-s', "' '",
+							'|', 'cut', "-d' '", '-f6'
+					],
+					this.link,
+			).then(output => output === '1' );
+	}
+
+	async testBootloaderIntegrity() {
+		return this.worker.executeCommandInHostOS(
+			['balena', 'run', '--rm', '-v', '/mnt:/mnt', 'alpine', '/bin/sh', '-c',
+				'"apk add --update --no-cache binutils && strip /mnt/boot/EFI/BOOT/bootx64.efi"'],
+			this.link,
+		).then(() => this.worker.executeCommandInHostOS('reboot', this.link)
+		).then(() => this.test.resolves(
+			this.waitForSerialOutput(/bootx64.efi: Access Denied/),
+			'Firmware will not load bootloader that fails verification',
+		)
+		).then( () => this.resetDUT() );
+	};
+
+	async testBootloaderConfigIntegrity() {
+		return this.worker.executeCommandInHostOS(
+			['sed', '-i', 's/lockdown=integrity//', '/mnt/efi/EFI/BOOT/grub.cfg'],
+			this.link
+		).then(() => this.worker.executeCommandInHostOS('reboot', this.link)
+		).then(() => this.test.resolves(
+			/* The below pattern is the expected output when the config file
+			 * fails the signature check and the fallback console is not enabled.
+			 *
+			 * This sequence of ANSI escape codes decodes as:
+			 *
+			 *	ESC[0m - reset all modes
+			 *	ESC[37m - white foreground
+			 *	ESC[40m - black background
+			 *
+			 *	https://gist.github.com/fnky/458719343aabd01cfb17a3a4f7296797
+			 *
+			 * Otherwise, the output finishes the above background color escape
+			 * sequence, clears the screen, and shows 'loading Boot0003
+			 * "balenaOS"'
+			 */
+
+			this.waitForSerialOutput(/\x1B\[0m\x1B\[37m\x1B\[40m$/, 3), // eslint-disable-line no-control-regex
+			'Bootloader will not load configuration that fails signature verification',
+			{ todo: 'This needs reworked for GRUB 2.12' },
+		)
+		).then( () => this.resetDUT() );
+	};
+}
+
+class rpiSecureBoot extends secureBoot {
+	async isSecureBootSupported() {
+		/* If the partition is split and there are boot.img and boot.sig */
```

Comment:
just checking - is this a Todo?

Reply:
Just a left over from when I was splitting tests - on x86 there is a difference between SB supported & enabled - I was wondering whether it made sense to test also for SB supported in RPI but I don't think so. If it's enabled it's supported anyway so this check would be redundant anyway.

I'll remove the comment, thanks.